actors/miner/tests/util.rs

```diff
@@ -204,8 +261,1024 @@ impl ActorHarness {
+                RawBytes::serialize(BigIntSer(&pledge_delta)).unwrap(),
+                TokenAmount::from(0),
+                RawBytes::default(),
+                ExitCode::Ok,
+            );
+        }
+
+        // Re-enrollment for next period.
+        if !cfg.no_enrollment {
+            let params = make_deadline_cron_event_params(cfg.expected_enrollment);
+            rt.expect_send(
+                *STORAGE_POWER_ACTOR_ADDR,
+                PowerMethod::EnrollCronEvent as u64,
+                RawBytes::serialize(params).unwrap(),
+                TokenAmount::from(0),
+                RawBytes::default(),
+                ExitCode::Ok,
+            );
+        }
+
+        let params = make_deferred_cron_event_params(
+            self.epoch_reward_smooth.clone(),
+            self.epoch_qa_power_smooth.clone(),
+        );
+        rt.set_caller(*POWER_ACTOR_CODE_ID, *STORAGE_POWER_ACTOR_ADDR);
+        rt.call::<Actor>(Method::OnDeferredCronEvent as u64, &RawBytes::serialize(params).unwrap())
+            .unwrap();
+        rt.verify();
+    }
+
+    pub fn submit_window_post(
+        &self,
+        rt: &mut MockRuntime,
+        deadline: &DeadlineInfo,
+        partitions: Vec<PoStPartition>,
+        infos: Vec<SectorOnChainInfo>,
+        cfg: PoStConfig,
+    ) {
+        let params = SubmitWindowedPoStParams {
+            deadline: deadline.index,
+            partitions: partitions,
+            proofs: make_post_proofs(self.window_post_proof_type),
+            chain_commit_epoch: deadline.challenge,
+            chain_commit_rand: Randomness(Vec::from(b"chaincommitment".clone())),
+        };
+        self.submit_window_post_raw(rt, deadline, infos, params, cfg).unwrap();
+        rt.verify();
+    }
+
+    fn submit_window_post_raw(
+        &self,
+        rt: &mut MockRuntime,
+        deadline: &DeadlineInfo,
+        infos: Vec<SectorOnChainInfo>,
+        params: SubmitWindowedPoStParams,
+        cfg: PoStConfig,
+    ) -> Result<RawBytes, ActorError> {
+        rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, self.worker);
+        let chain_commit_rand = match cfg.chain_randomness {
+            Some(r) => r,
+            None => params.chain_commit_rand.clone(),
+        };
+        rt.expect_get_randomness_from_tickets(
+            DomainSeparationTag::PoStChainCommit,
+            params.chain_commit_epoch,
+            Vec::new(),
+            chain_commit_rand,
+        );
+        rt.expect_validate_caller_addr(self.caller_addrs());
+
+        let challenge_rand = Randomness(Vec::from([10, 11, 12, 13]));
+
+        // only sectors that are not skipped and not existing non-recovered faults will be verified
+        let mut all_ignored = BitField::new();
+        let mut all_recovered = BitField::new();
+        let dln = self.get_deadline(rt, deadline.index);
+        for p in &params.partitions {
+            let partition = dln.load_partition(&rt.store, p.index).unwrap();
```

Comment:
Another spurious case handled differently.  In go actors a non-existing partition will continue the loop whereas here the whole test will fail.  In this case I think your change is a better way to handle it.

Reply:
yeah, the go code made no sense in that case, so i took a little liberty.